satpy/readers/nwcsaf_hrw_nc.py

```diff
@@ -0,0 +1,331 @@
+        self.platform_name = PLATFORM_NAMES.get(self.h5f.attrs["satellite_identifier"].astype(str))
+        self.sensor = SENSOR.get(self.platform_name, "seviri")
+        self.lons = {}
+        self.lats = {}
+        # Imaging period, which is set after reading any data, and used to calculate end time
+        self.period = None
+
+        # The resolution is given in kilometers, convert to meters
+        self.resolution = 1000 * self.h5f.attrs["spatial_resolution"].item()
+
+    def __del__(self):
+        """Close file handlers when we are done."""
+        with suppress(OSError):
+            self.h5f.close()
+
+    def available_datasets(self, configured_datasets=None):
+        """Form the names for the available datasets."""
+        for channel in WIND_CHANNELS:
+            prefix = self._get_channel_prefix(channel)
+            dset = self.h5f[channel]
+            for measurand in dset.dtype.fields.keys():
+                if measurand == "trajectory":
+                    continue
+                ds_info = self._measurand_ds_info(prefix, measurand)
+                yield True, ds_info
+            if self.merge_channels:
+                break
+
+    def _get_channel_prefix(self, channel):
+        if self.merge_channels:
+            return ""
+        return channel + "_"
+
+    def _measurand_ds_info(self, prefix, measurand):
+        ds_info = {
+            "file_type": self.filetype_info["file_type"],
+            "resolution": self.resolution,
+            "name": prefix + measurand,
+        }
+        if measurand not in ("longitude", "latitude"):
+            ds_info["coordinates"] = (prefix + "longitude", prefix + "latitude")
+        if measurand == "longitude":
+            ds_info["standard_name"] = "longitude"
+        if measurand == "latitude":
+            ds_info["standard_name"] = "latitude"
+
+        return ds_info
+
+    def get_dataset(self, key, info):
+        """Load a dataset."""
+        logger.debug("Reading %s.", key["name"])
+        if self.merge_channels:
+            data = self._read_merged_dataset(key)
+        else:
+            data = self._read_dataset(key)
+        data.attrs.update(info)
+
+        return data
+
+    def _read_merged_dataset(self, dataset_key):
+        """Read a dataset merged from every channel."""
+        dataset_name = dataset_key["name"]
+
+        data = []
+        collect_coords = True
+        if "merged" in self.lons:
+            collect_coords = False
+        for channel in WIND_CHANNELS:
+            if collect_coords:
+                self._read_channel_coordinates(channel)
+                self._append_merged_coordinates(channel)
+            if self.period is None:
+                self.period = self.h5f[channel].attrs["time_period"].item()
+            try:
+                data.append(self.h5f[channel][dataset_name])
+            except ValueError:
+                logger.warning("Reading %s is not supported.", dataset_name)
```

Comment:
I would have thought this raises a KeyError...

Reply:
It raises `ValueError` when the data are there but the compound datatype is unreadable.